inductiva/resources/machines_base.py

```diff
@@ -0,0 +1,214 @@
+"""Base class for machine groups."""
+import time
+
+from absl import logging
+
+import inductiva
+import inductiva.client.models
+from inductiva import api
+from inductiva.client.apis.tags import instance_api
+
+
+class BaseMachineGroup():
+    """Base class to manage Google Cloud resources."""
+    def __init__(
+        self,
+        machine_type: str,
+        spot: bool = False,
+        disk_size_gb: int = 40,
+        zone: str = "europe-west1-b"
+    ) -> None:
+        """Create a MachineGroup object.
+
+        Args:
+            machine_type: The type of GC machine to launch. Ex: "e2-standard-4".
+            Check https://cloud.google.com/compute/docs/machine-resource for
+            more information about machine types.
+            num_machines: The number of virtual machines to launch.
+            spot: Whether to use spot machines.
+            disk_size_gb: The size of the disk in GB, recommended min. is 40 GB.
+            zone: The zone where the machines will be launched.
+        """
+        self.machine_type = machine_type
+        self.spot = spot
+        self.disk_size_gb = disk_size_gb
+        self.zone = zone
+        self._id = None
+        self._name = None
+        self.create_time = None
+        self._started = False
```

Comment:
Yes, and that's using the @property decorator that I used a few lines below. But I only used it for id and name since they are the most important parameters, I think, that the used cannot alter but can only view.